### ISSUE TITLE
Error in class info generation!

### ISSUE THREAD

**Issue by davidyee** [canoo/dolphin-platform#370]
Hi, I receive the following error on the client-end regarding a ClassNotFoundException with a JavaFX + Spring Boot project using Dolphin Platform 0.8.13. There are no visible errors on the server side.

```
[WARNING]
java.lang.RuntimeException: Error in class info generation!
	at com.canoo.dolphin.impl.ClassRepositoryImpl$1.modelStoreChanged(ClassRepositoryImpl.java:71)
	at org.opendolphin.core.ModelStoreListenerWrapper.modelStoreChanged(ModelStoreListenerWrapper.java:57)
	at org.opendolphin.core.ModelStore.fireModelStoreChangedEvent(ModelStore.java:344)
	at org.opendolphin.core.ModelStore.add(ModelStore.java:137)
	at org.opendolphin.core.client.ClientModelStore.add(ClientModelStore.java:66)
	at org.opendolphin.core.client.comm.ClientResponseHandler.handleCreatePresentationModelCommand(ClientResponseHandler.java:116)
	at org.opendolphin.core.client.comm.ClientResponseHandler.dispatchHandle(ClientResponseHandler.java:53)
	at org.opendolphin.core.client.comm.AbstractClientConnector.dispatchHandle(AbstractClientConnector.java:202)
	at org.opendolphin.core.client.comm.AbstractClientConnector.processResults(AbstractClientConnector.java:170)
	at org.opendolphin.core.client.comm.AbstractClientConnector$2$1$1.run(AbstractClientConnector.java:121)
	at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: com.example.MyDto
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.canoo.dolphin.impl.ClassRepositoryImpl$1.modelStoreChanged(ClassRepositoryImpl.java:65)
	... 16 more
```

The `com.example.MyDto` is a `@DolphinBean` which contains other `@DolphinBean` objects. I can see the class file for `com.example.MyDto` so it is not a missing file problem.

If I try to open the same window that generated the above error (still on the client side), I receive a different error:

```
[WARNING] 
java.lang.NullPointerException: Argument 'classInfo' may not be null
	at java.util.Objects.requireNonNull(Objects.java:228)
	at com.canoo.dolphin.util.Assert.requireNonNull(Assert.java:45)
	at com.canoo.dolphin.impl.AbstractBeanBuilder$1.onEvent(AbstractBeanBuilder.java:61)
	at com.canoo.dolphin.impl.EventDispatcherImpl.onAddedHandler(EventDispatcherImpl.java:109)
	at com.canoo.dolphin.impl.EventDispatcherImpl.modelStoreChanged(EventDispatcherImpl.java:77)
	at org.opendolphin.core.ModelStoreListenerWrapper.modelStoreChanged(ModelStoreListenerWrapper.java:57)
	at org.opendolphin.core.ModelStore.fireModelStoreChangedEvent(ModelStore.java:344)
	at org.opendolphin.core.ModelStore.add(ModelStore.java:137)
	at org.opendolphin.core.client.ClientModelStore.add(ClientModelStore.java:66)
	at org.opendolphin.core.client.comm.ClientResponseHandler.handleCreatePresentationModelCommand(ClientResponseHandler.java:116)
	at org.opendolphin.core.client.comm.ClientResponseHandler.dispatchHandle(ClientResponseHandler.java:53)
	at org.opendolphin.core.client.comm.AbstractClientConnector.dispatchHandle(AbstractClientConnector.java:202)
	at org.opendolphin.core.client.comm.AbstractClientConnector.processResults(AbstractClientConnector.java:170)
	at org.opendolphin.core.client.comm.AbstractClientConnector$2$1$1.run(AbstractClientConnector.java:121)
	at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
	at java.lang.Thread.run(Thread.java:745)
```

Is there anything I should look at in order to isolate the cause of this problem? I'm not sure if the following is related, but I have also recently noticed that I cannot run my JavaFX application directly in Eclipse (the program terminates immediately) but instead have to use `mvn compile exec:java`.

**Comment by hendrikebbers** [comment]
Hi,

for me the eclipse problem and the CNFE may be results based on the same problem. Maybe something in your maven setup is wrong so that the module that contains the MyDTO class is not on the class path. You can simply check this by creating a MyDTO instance by hand directly in the main method of the client:

public static void main(...) { MyDto instance = new MyDto(); }

If it's not on the class path a CNFE will be thrown, too. Otherwise this is something different and we can have a deeper look.

**Comment by davidyee** [comment]
Thanks for the suggestion. I double checked to see if I could see the `common` jar in Maven and renamed it to something new like `common2` to test if it was being updated and created properly. I setup Eclipse to use a different Maven installation and I found that I needed to run `mvn install` to make sure that the new `common2` showed up after which I was able to use the Dolphin Bean without the CNFE error. 

Maybe it's because I am still learning to use Maven but it could be helpful to add that the `common` package should have the `install` goal added to make sure updated versions get added to the local repository (where they later get picked up and used by the client and server). At least that is the case when running the server and client in Maven through Eclipse. I think I didn't encounter this problem before because I could run the ClientApplication as a Java Application in Eclipse. And all of that said, now it doesn't seem that the original problem I had with Eclipse not running the ClientApplication has too much to do with this CNFE problem.

That said, I can't replicate this behaviour on my other computer so it may be a problem with my Eclipse workspace.
